jobs/FindLabeledImagesInResults.php

```diff
@@ -0,0 +1,171 @@
+        $this->db = new mysqli(
+            $config['db']['host'],
+            $config['client']['user'],
+            $config['client']['password'], $config['db']['dbname']
+        );
+        if ( $this->db->connect_error ) {
+            die('DB connection Error (' . $this->db->connect_errno . ') '
+                . $this->db->connect_error);
+        }
+
+        $this->searchUrl = $config['search']['baseUrl'];
+
+        $this->ch = curl_init();
+        curl_setopt( $this->ch, CURLOPT_RETURNTRANSFER, true );
+
+        $this->log = fopen(
+            __DIR__ . '/../' . $config['log']['findLabeledImages'],
+            'a'
+        );
+
+        $this->description = $description;
+        if ( $this->description == '' ) {
+            $this->description = 'Search of ' . $this->searchUrl . ' on ' . date('Y-m-d H:i:s');
+        }
+    }
+
+    public function __destruct() {
+        curl_close( $this->ch );
+        fclose( $this->log );
+    }
+
+    public function run() {
+        $this->log( 'Begin ' . $this->description );
+        $searchId = $this->createSearchRecord();
+        foreach ( $this->getSearchTerms() as $searchTerm ) {
+            $this->log( 'Searching ' . $searchTerm );
+            $searchTerm = trim( $searchTerm );
+            $this->processResults( $searchTerm, $this->search( $searchTerm ), $searchId );
+        }
+        $this->log( 'End ' . $this->description );
+        return $searchId;
+    }
+
+    public function log( string $msg ) {
+        fwrite( $this->log, date( 'Y-m-d H:i:s' ) . ': ' . $msg . "\n" );
+    }
+
+    private function createSearchRecord() {
+        $this->db->query(
+            'insert into search set ' .
+            'description="' . $this->db->real_escape_string( $this->description ). '"'
+        );
+        return $this->db->insert_id;
+    }
+
+    private function getSearchTerms() : array {
+        $searchTerms = [];
+        $searchTermResults = $this->db->query(
+            'select distinct(term) as term from results_by_component where rating is not null'
+        );
+        while ( $searchTerm = $searchTermResults->fetch_assoc() ) {
+            $searchTerms[] = $searchTerm['term'];
+        }
+        return $searchTerms;
+    }
+
+    private function search( string $searchTerm ) : array {
+        curl_setopt( $this->ch, CURLOPT_URL, $this->getSearchUrl( $searchTerm ) );
+        $result = curl_exec( $this->ch );
+        if ( curl_errno( $this->ch ) ) {
+            $this->log( curl_error( $this->ch ) . ':' . curl_errno( $this->ch ) );
+            die( 'Exiting because of curl error, see log for details.' );
+        }
+        $array = json_decode( $result, true );
+        return $array;
+    }
+
+    private function getSearchUrl( string $searchTerm ) : string {
+        return sprintf(
+            $this->searchUrl . '/w/index.php?search=%s+filetype:bitmap&ns6=1&' .
+            'cirrusDumpResult&mediasearch=1&limit=100',
```

Comment:
This should be a massive number. I originally thought slightly-over-100 would suffice, but it doesn't, not even close.
The original data was gathered via many different scoring mechanisms - some of which found (relevant) files that only have a score via a field that usually has a very low weight & contributes little towards the final score (but it did end up in the dataset because the we ranked solely for that field)
As a result, we have a a whole bunch of images that - with our default set of weights - won't ever be found near the top 100 (or even 100000, for popular terms) results. It's not that the algorithms doesn't find them (recall), but they're being flooded out by other relevant matches. For decent recall calculation, we'll need pretty much the entire resultset, not a limit...